jena-arq/src/main/java/org/apache/jena/query/Dataset.java

```diff
@@ -113,4 +113,11 @@
      *  The dataset can not be used for query after this call.
      */
     public void close() ;
+    
+    /**
+     * @return Whether this {@code Dataset} is empty of graphs. Be aware of the semantic looseness inherent in
+     * <a href="https://www.w3.org/TR/2014/REC-rdf11-concepts-20140225/#h_note_4">the definition
+     * of RDF Datasets</a>; whether a named graph exists if nothing is in it is implementation-specific.
+     */
+    boolean isEmpty();
```

Comment:
"is empty of graphs" 

Firstly, there is always at least one graph (the default graph).

Secondly, "isEmpty" on a DSG means no triples in any graph (default implement is `!contain(ANY,ANY,ANY,ANY)` which includes the default graph).

Reply:
Oh, fudge. Then we really can't have a default impl of this, can we?

Reply:
Sure we can.  Change the javadoc to "returns true if the dataset contains no statements in any model". That's what it does, and what the test does anyway, because the impl passes down to `DatasetGraph.isEmpty`. That is the useful operation.

Reply:
I didn't do that at first because it felt like a bit of a conflict against the rest of the API for `Dataset`, which discusses models/graphs and not tuples. But if you're okay with it, it doesn't bother me.